### ISSUE TITLE
az ad sp credential list crash on mac

### ISSUE THREAD

**Issue by jlpedrosa** [Azure/azure-cli#9248]
Hi 

## Describe the bug
Az crash when running list credentials of a valid service principal

**Command Name**
`az ad sp credential list`

**Errors:**
```The command failed with an unexpected error. Here is the traceback:

Parameter 'application_object_id' can not be None.Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/role/commands.py", line 69, in graph_err_handler
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/role/custom.py", line 1085, in list_service_principal_credentials
    return _get_service_principal_credentials(graph_client, app_object_id, cert)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/role/custom.py", line 1092, in _get_service_principal_credentials
    app_creds = list(graph_client.applications.list_password_credentials(app_object_id))
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/msrest/paging.py", line 143, in __next__
    self.advance_page()
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/msrest/paging.py", line 129, in advance_page
    self._response = self._get_next(self.next_link)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/graphrbac/operations/applications_operations.py", line 669, in internal_paging
    'applicationObjectId': self._serialize.url("application_object_id", application_object_id, 'str'),
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/msrest/serialization.py", line 592, in url
    data = self.validate(data, name, required=True, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/msrest/serialization.py", line 662, in validate
    raise ValidationError("required", name, True)
msrest.exceptions.ValidationError: Parameter 'application_object_id' can not be None.
```

## To Reproduce:
Steps to reproduce the behavior. Note: Command arguments have been redacted.
Having a valid service principal:
- `Run: az ad sp credential list --id {}`

## Expected Behavior

The Az CLI not crashing and even provide data about the SP :)

## Environment Summary
```
Darwin-18.5.0-x86_64-i386-64bit
Python 3.7.3
Shell: bash

azure-cli                         2.0.62 *
azure-cli-iot-ext                  0.7.0

Extensions:
azure-cli-iot-ext                  0.7.0
aks-preview                        0.3.1

Python location '/usr/local/Cellar/azure-cli/2.0.62/libexec/bin/python'
Extensions directory '/Users/jopedros/.azure/cliextensions'

Python (Darwin) 3.7.3 (default, Mar 27 2019, 09:23:15) 
[Clang 10.0.1 (clang-1001.0.46.3)]

```

**Comment by tjprescott** [comment]
A fix has already been submitted so that a stack trace will not thrown due to an SDK's validation error, but the question remains why application ID is none.

**Comment by jlpedrosa** [comment]
> A fix has already been submitted so that a stack trace will not thrown due to an SDK's validation error, but the question remains why application ID is none.

Hi @tjprescott  

I've found the issue. This behaviour happens when the given application id does not contain any secrets. Then you'd get this error, while if the application contains some secrects they get listed.

JL

**Comment by jlpedrosa** [comment]
Hi!

Any feedback?

**Comment by tjprescott** [comment]
@marstr @yugangw-msft for comment.

**Comment by marstr** [comment]
Does this repro on 2.0.67, @jlpedrosa? I know @yugangw-msft has been chasing down bugs due to this in many commands in the last versions. It's definitely worth seeing if it still happens.

**Comment by yugangw-msft** [comment]
#9337 should fix it, but is blocked by the #9747. We am trying to get both in by tomorrow.

**Comment by yugangw-msft** [comment]
#9337 merged
